test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -725,6 +725,32 @@ TEST_P(ExtAuthzHttpIntegrationTest, DefaultCaseSensitiveStringMatcher) {
   ASSERT_TRUE(header_entry.empty());
 }
 
+TEST_P(ExtAuthzHttpIntegrationTest, DeniedResponseWhenAuthzReturn5xx) {
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.envoy.reloadable_features.ext_authz_"
+                                    "http_service_5xx_is_denied_instead_of_error",
+                                    "false");
+  initializeConfig();
+  HttpIntegrationTest::initialize();
+  initiateClientConnection();
+
+  AssertionResult result =
+      fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, fake_ext_authz_connection_);
+  RELEASE_ASSERT(result, result.message());
+  FakeStreamPtr ext_authz_request;
+  result = fake_ext_authz_connection_->waitForNewStream(*dispatcher_, ext_authz_request);
+  RELEASE_ASSERT(result, result.message());
+  result = ext_authz_request->waitForEndStream(*dispatcher_);
+  RELEASE_ASSERT(result, result.message());
+
+  Http::TestResponseHeaderMapImpl ext_authz_response_headers{{":status", "500"}};
+  ext_authz_request->encodeHeaders(ext_authz_response_headers, true);
```

Comment:
Can we do `encodeData` as well? Hence we are sure the body from ext_authz service is returned back to the client.

Reply:
i tried to add this, but I cant make the test works. any pointer that I can follow for this?